alc/backends/wasapi.cpp

```diff
@@ -1232,28 +1232,49 @@ FORCE_ALIGN int WasapiPlayback::mixerProc()
     auto avhandle = AvrtHandlePtr{AvSetMmThreadCharacteristicsW(taskname, &sAvIndex)};
 #endif
 
+    const auto sharemode =
+        GetConfigValueBool(mDevice->mDeviceName, "wasapi", "exclusive-mode", false)
+        ? AUDCLNT_SHAREMODE_EXCLUSIVE : AUDCLNT_SHAREMODE_SHARED;
+
     mBufferFilled = 0;
     while(!mKillNow.load(std::memory_order_relaxed))
     {
+        DWORD res{ WaitForSingleObjectEx(mNotifyEvent, 2000, FALSE) };
+        if (res != WAIT_OBJECT_0)
+            ERR("WaitForSingleObjectEx error: {:#x}", res);
+
         UINT32 written;
-        HRESULT hr{audio.mClient->GetCurrentPadding(&written)};
-        if(FAILED(hr))
+        uint len = 0;
+        HRESULT hr = S_FALSE;
+        if(sharemode == AUDCLNT_SHAREMODE_EXCLUSIVE)
         {
-            ERR("Failed to get padding: {:#x}", as_unsigned(hr));
-            mDevice->handleDisconnect("Failed to retrieve buffer padding: {:#x}",
-                as_unsigned(hr));
-            break;
+            hr = audio.mClient->GetBufferSize(&written);
```

Comment:
Unless this value can change, it was already stored in `mOutBufferSize`, which is copied into the local variable `buffer_len`. It doesn't need to be queried each time. Also, since `written` is used to update `mPadding`, which is used for latency calculations, I'm not sure leaving it as the buffer size is a good idea (unless that is the actual amount of padding WASAPI holds for exclusive streams, i.e. the amount of audio still left to play before the new samples to be mixed).